src/Mantasflowers.WebApi/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Mantasflowers.Contracts.Errors;
+using Mantasflowers.Contracts.Firebase.Request;
+using Mantasflowers.Contracts.Firebase.Response;
+using Mantasflowers.Contracts.User.Response;
+using Mantasflowers.Services.FirebaseService;
+using Mantasflowers.Services.ServiceAgents.Exceptions;
+using Mantasflowers.Services.Services.User;
+using Mantasflowers.WebApi.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.WebApi.Controllers
+{
+    [ApiController]
+    [Route("/authentication")]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(string), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status500InternalServerError)]
+    public class AuthenticationController : ControllerBase
+    {
+        private readonly FirebaseService _fbService;
+        private readonly IUserService _userService;
+
+        public AuthenticationController(FirebaseService fbService, IUserService userService)
+        {
+            _fbService = fbService;
+            _userService = userService;
+        }
+
+        [HttpPost("login")]
+        [ProducesResponseType(typeof(PostTokensResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> LoginAsync([FromBody] PostCredentialsRequest request)
+        {
+            try
+            {
+                var response = await _fbService.GetTokensAsync(request);
+                return Ok(response);
+            }
+            catch (FirebaseTokenException e)
+            {
+                return Unauthorized(e.Message);
+            }
+        }
+
+        [HttpPost("refresh-token")]
```

Comment:
> 
> 
> I had a thought about `refresh-token` endpoint. Maybe we should also delete the current refresh token in the process? @alwaysintune

What do you mean exactly? This endpoint does not return a new refresh token.

Reply:
I mean that while the intended flow is:
1. Get normal token + refresh token.
2. Normal token expires.
3. Send refresh token.
4. Get new normal token + refresh token.

After step 4 the previous refresh token is still valid. As such, you can generate many of them. I don't know enough about JWT judge whether it's an issue though

Reply:
And it does return a new refresh token 🕺 [as seen here](https://github.com/mantasflowers/Mantasflowers.Backend/blob/auth-structure-refactor/src/Mantasflowers.WebApi/Controllers/AuthenticationController.cs#L51) 🕺

Reply:
Refresh tokens from step 3. and 4. are identical.

Reply:
Just tested, returned refresh token is the same as in a request.

Reply:
I read up on documentation a bit. My bad. I do think we should communicate over the fact that refresh token will expire as well and frontend will eventually have to call /login again. Feel free to resolve this.